### DISCUSSION TITLE
Navbar: Nesting children does not work with 3 levels or more

### DISCUSSION THREAD

**Discussion by Sennik** [statamic/cms#5769] [Q&A] [answered]
Pulling my hair out here. Statamic 3.3.4: Trying to create a muti level navbar from a collection. This works until I reach the third (green) level, then it fails: 

`syntax error, unexpected token "endif", expecting end of file on line 7 of: <h1>{{ title }}</h1> <?php if (false): ?> tag_pair_2749b4ee46c1138e5c1ccb1a21498a6b <?php endif; ?> {{ /children }} <?php endif; ?> `

![afbeelding](https://user-images.githubusercontent.com/5636394/161992877-4182029e-2fc5-4bb5-8b08-fb1dc9a5c239.png)

**The (simplified) code`, this code fails:**

```
{{ nav:collection:pages }}
    <h1>{{ title }}</h1>
    {{ if children }}
        {{ children }}
            <h2>{{ title }}</h2>
            {{ if children }}
                {{ children }}
                    <h3>{{ title }}</h3>
                {{ /children }}
            {{ /if }}
        {{ /children }}
    {{ /if }}
{{ /nav:collection:pages }}
```
**If I remove the third inner loop:**
```
{{ children }}
    <h3>{{ title }}</h3>
{{ /children }}
```
The code runs without error but of course without the third level I need.

**Comment by jasonvarga** [comment, marked as answer]
Sounds like an issue with the old parser. I think if you use the new one, this should be solved.

https://statamic.dev/new-antlers-parser

**Reply by Sennik** [reply]
Sure, but that opens another can of worms, besides the website advises to use this new version on new websites. Is the new parser the only road to go?

**Reply by jasonvarga** [reply]
Some bugs we consider fixed in the new parser and won't be done in the old one. This would be one of them.

If you really don't want to use the new parser, you can move your nested `children` tag pair into a partial.

What's happening is that the parser is being confused by two closing `{{ /children }}` tags.

**Reply by Sennik** [reply]
I was using 1 partiel already, still didn;t work.

But I bit the bullet and started using the new parser. Problem there was that I failed on some of my messy coding (Using variables in Glide tags) but then again, time to clean up. It works now! Thanks for your support and of course, your great work on Statamic!
